src/openrct2/interface/widget.c

```diff
@@ -1105,6 +1105,7 @@ static void widget_text_box_draw(rct_drawpixelinfo *dpi, rct_window *w, rct_widg
 
 	// Make a copy of the string for measuring the width.
 	char temp_string[TEXT_INPUT_SIZE] = { 0 };
+	gTextInput = context_start_text_input(gTextBoxInput, TEXT_INPUT_SIZE);
```

Comment:
I don't think it should start a new text input session, it is supposed to be drawing the caret for the existing one.